bin/test/main.ml

```diff
@@ -371,8 +380,15 @@ let run setup non_deterministic not_verbose syntax silent verbose_findlib
     try
     run_exn setup non_deterministic not_verbose syntax silent verbose_findlib
       prelude prelude_str file section root direction force_output
-    with Failure f -> (prerr_endline f; exit 1)
- 
+    with
+    | Failure f ->
+      prerr_endline f;
+      1
+    | Test_block_failure (block, msg) ->
+      Fmt.epr "Error in block at line %d in %s:@\n%s\n"
```

Comment:
I think the error message could be slightly improved here by:
- Using `code block` instead of block, I think it sounds a bit vague
- Adding the kind of block, OCaml, OCaml toplevel or shell
- printing locations the other way around, file first, line second

Reply:
Changed it. What do you think?